kolena/_utils/datatypes.py

```diff
@@ -77,13 +79,44 @@ def _allow_extra(cls: Type[T]) -> bool:
 _DATA_TYPE_MAP: Dict[str, Type["TypedDataObject"]] = {}
 
 
+class DataCategory(str, Enum):
+    TEST_SAMPLE = "TEST_SAMPLE"
+    PLOT = "PLOT"
+    METRICS = "METRICS"
+    ASSET = "ASSET"
+    ANNOTATION = "ANNOTATION"
+
+    def data_category_to_module_name(self) -> str:
+        if self == DataCategory.TEST_SAMPLE:
+            return "kolena.workflow.testing"
+        if self == DataCategory.PLOT:
+            return "kolena.workflow.plot"
+        if self == DataCategory.METRICS:
+            return "kolena._experimental.workflow.thresholded"
+        if self == DataCategory.ASSET:
+            return "kolena.asset"
+        if self == DataCategory.ANNOTATION:
+            return "kolena.annotation"
+        raise ValueError(f"Must specify module name for data category: {self}")
+
+
 def _get_full_type(obj: Type["TypedDataObject"]) -> str:
     data_type = obj._data_type()
-    return f"{data_type._data_category()}/{data_type.value}"
+    return f"{data_type._data_category().value}/{data_type.value}"
 
 
 def _get_data_type(name: str) -> Optional[Type["TypedDataObject"]]:
-    return _DATA_TYPE_MAP.get(name, None)
+    class_type = _DATA_TYPE_MAP.get(name, None)
```

Comment:
Can we add a test to make sure the original bug will have been fixed?

Reply:
yes will add a test